stacks_queues/queue.py

```diff
@@ -23,40 +23,82 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if (self.front == 0 and self.rear == self.buffer_size - 1) or (self.rear == (self.front - 1)%(self.buffer_size -1)):
+            raise QueueFullException
+        
+        elif self.front == -1:
+            self.front = 0 
+            self.rear = 0 
+            self.store[0] = element
+            self.size += 1
+        elif self.rear == self.buffer_size - 1 and self.front != 0: 
+            self.rear = 0
+            self.store[0] = element
+            self.size += 1
+        else:
+            self.rear += 1
+            self.store[self.rear] = element
+            self.size += 1
+
+        return None
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException
+
+        element = self.store[self.front]
+        self.store[self.front] = None
+        self.front += 1
+        self.size -= 1
+            
+        if self.front == len(self.store):
+            self.front = 0 
+            
```

Comment:
Take a look at `dequeue` and see how this can be made redundant with the `%` operator.